### ISSUE TITLE
Drop to tree root

### ISSUE THREAD

**Issue by torava** [reactabular/reactabular#328]
Based on the example in the documentation, I'm changing Reactabular tree's drag and drop behavior so that dragged item is inserted as children to the target item instead of changing the order. This is the same behavior than for example in file browsers. I faced difficulties in the use case in which dragged item is moved to tree root. Like in file browsers, tree can have multiple root items. In that case item would be moved to container. The problem is that reactabular dnd allows drag and drop between items but not between item and container.

Proposal:

Reactabular should catch also drag and drop between item and container so that additional functions could be implemented for it as well.

With this feature, you should also provide a file-explorer-like example for dnd tree in documentation. I think it is as important as ordering a tree and I feel that many projects could benefit from that. It's something on which I don't expect to get my hands dirty since I'm using a library. You should also support modularity and allow abstracting dnd tree. Dnd tree is quite a general element but with reactabular I find it hard to follow DRY principle. For instance, since toggleChildren is implemented in column definition and refers to this.state.rows, it's not possible to use the same DragAndDropTree component in multiple higher level components with different columns. Instead, I need to repeat DragAndDropTree component multiple times.

**Comment by bebraw** [comment]
Yeah, I can accept a PR for this if someone wants to give a go. Note that I don't do active development anymore due to a lack of funding but I don't mind merging PRs and helping with reviews.
